### DISCUSSION TITLE
COM port issue?

### DISCUSSION THREAD

**Discussion by novon1** [mathoudebine/turing-smart-screen-python#438] [Q&A]
Total novice here. I was able to get it running on python 3.12 in my windows 11 pc. And I am pretty sure my device is the XuanFang 3.5" flagship as per this link: https://github.com/mathoudebine/turing-smart-screen-python/wiki/Hardware-revisions.
However, when I go to configure, in the COM port drop-down doesn't show any option other than Auto detection. I was able to force it use COM3 and COM10 by editing the config.yaml file but I am still getting the error: Cannot open COM port.

1/6/2024 7:16:14 PM [INFO] Loading theme LandscapeMagicBlue from res/themes/LandscapeMagicBlue/theme.yaml
1/6/2024 7:16:14 PM [DEBUG] HW revision: B
1/6/2024 7:16:14 PM [ERROR] Cannot find COM port automatically, please run Configuration again and select COM port manually
1/6/2024 7:17:44 PM [INFO] Loading theme LandscapeMagicBlue from res/themes/LandscapeMagicBlue/theme.yaml
1/6/2024 7:17:44 PM [DEBUG] HW revision: B
1/6/2024 7:17:44 PM [DEBUG] Static COM port: COM3
1/6/2024 7:17:44 PM [ERROR] Cannot open COM port COM3: could not open port 'COM3': FileNotFoundError(2, 'The system cannot find the file specified.', None, 2)
1/6/2024 7:18:06 PM [INFO] Loading theme LandscapeMagicBlue from res/themes/LandscapeMagicBlue/theme.yaml
1/6/2024 7:18:06 PM [DEBUG] HW revision: B
1/6/2024 7:18:06 PM [DEBUG] Static COM port: COM10
1/6/2024 7:18:06 PM [ERROR] Cannot open COM port COM10: could not open port 'COM10': FileNotFoundError(2, 'The system cannot find the file specified.', None, 2)

Any ideas what I am doing wrong? Sorry for the big post.

**Comment by Piro3202** [comment]
It is works normally on stock application? And the screen is visible in Device Manager as com device on specified port? In my pc it shows here as USB Serial Device at COM3. Serial communication at newer windows is always bit tricky.
![Zrzut ekranu 2024-01-07 055837](https://github.com/mathoudebine/turing-smart-screen-python/assets/136054148/3bc4a168-fd4a-4972-9798-ddec21144795)

**Reply by novon1** [reply]
Tried running the stock application "ExtendScreen v2.5" with no success.
Your screenshot actually helped. I had to manually install PORTS (COM & LPT) in device manager from a list of legacy hardware (win 11). Now getting this error.

1/7/2024 7:03:54 PM [INFO] Loading theme Landscape_simple_cyberpunk from res/themes/Landscape_simple_cyberpunk/theme.yaml
1/7/2024 7:03:54 PM [DEBUG] HW revision: B
1/7/2024 7:03:54 PM [DEBUG] Static COM port: COM3
1/7/2024 7:03:54 PM [ERROR] Cannot open COM port COM3: could not open port 'COM3': FileNotFoundError(2, 'The system cannot find the file specified.', None, 2)
![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/155824567/f4369c68-ac1b-428b-8cd7-824e0d11da13)
![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/155824567/13a91887-6e7f-43d1-a532-dfd4550024c2)

**Reply by mathoudebine** [reply]
@novon1 you do not have to install the COM ports manually from the Legacy Hardware menu, this is for old computers only with physical serial ports. You can remove them (COM3 and COM4).
The COM port should appear automatically in the list as soon as you plug the screen.
Can you try using another USB-C cable to connect your screen to your computer?
Keep the Device Manager open to check if the COM port appears as soon as the screen is plugged

**Reply by novon1** [reply]
Sorry for the late reply. No COM port is appearing. It shows up as an HID device.

![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/155824567/7f512d36-a9a5-4ec1-a0c9-d6823167bc04)

**Reply by mathoudebine** [reply]
Unfortunately, if it shows up as an HID device then it is not supported by this project. Only smart screens detected as serial ports are compatible. Can you tell me where you bought it so that I can help identify the exact model?

**Comment by Tailslide** [comment]
I'm having the same issue.. screen looks the same phsyically as your compatible list XuanFang 3.5" but is recognized as follows:

\\GPALIEN1\root\cimv2:Win32_PnPEntity.DeviceID="USB\\VID_0483&PID_0065\\2022100600"
\\GPALIEN1\root\cimv2:Win32_PnPEntity.DeviceID="HID\\VID_0483&PID_0065\\9&2EDA18E6&0&0000"

Found device: {'path': b'\\\\?\\HID#VID_0483&PID_0065#9&2eda18e6&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}', 'vendor_id': 1155, 'product_id': 101, 'serial_number': '2022100600', 'release_number': 2, 'manufacturer_string': 'llhmi.com', 'product_string': 'llTechCo.,Ltd     ', 'usage_page': 65280, 'usage': 1, 'interface_number': 0, 'bus_type': 1}

Bought from: 

https://www.aliexpress.com/item/1005005058181722.html

**Reply by mathoudebine** [reply]
You may have this type of screen:
![image](https://github.com/user-attachments/assets/7917cde7-6563-4712-8570-d78bf0e0889f)
Unfortunately it is not supported by this project.
Just to be sure, what is the software provided with the screen?
